src/pages/index.js

```diff
@@ -1,275 +1,233 @@
-const currentProfileName = document.querySelector('.profile__name');
-const currentProfileDescription = document.querySelector('.profile__description');
-//#endregion
+const formValidators = {};
+enableValidation(validationSettings);
 
-//#region new post modal declarations
-const defaultLinkField = 'Paste a link to the picture';
-const defaultCaptionField = 'Type your caption'
+const userConfirmDeleteModal = new ModalConfirm({
+  modalSelector: modalSelectors.confirmModalSelector,
+  confirmText: "Delete",
+  awaitingConfirmText: "Deleting...",
+});
 
-const postModal = document.querySelector('#post-modal');
-const postLink = postModal.querySelector('#link');
-const postCaption = postModal.querySelector('#caption');
-postModal.addEventListener('click',checkForModalOverlayClick);
-document.querySelector('.profile__add-button').addEventListener('click',onNewPostClick);
-document.querySelector('#new-post-modal__close').addEventListener('click',onCloseModalClick);
-const cardForm = document.forms['new-post-modal__form']
-cardForm.addEventListener('submit',onPostModalSubmit);
-//#endregion
+const cardSection = new Section({
+  containerSelector: `.cards__list`,
+  createItem: (data) => {
+    return makeNewCard(data);
+  },
+  initialize: (route) => {
+    dbApi
+      .getBatchData(route)
+      .then((data) => {
+        cardSection.createItems(data);
+        cardSection.renderItems();
+      })
+      .catch((err) => console.error(`Error initializing card section: ${err}`));
+  },
+});
 
-//#region preview modal declarations
-const previewModal = document.querySelector('#preview-modal');
-const previewImage = document.querySelector('.modal__preview-image');
-const previewTitle = document.querySelector('#preview-title');
-previewModal.addEventListener('click',checkForModalOverlayClick);
-document.querySelector('#preview-modal__close').addEventListener('click',onCloseModalClick);
-//#endregion
+cardSection.initialize(routeData.getCards);
+
+const userProfile = new UserProfile({
+  initialize: () => {
+    dbApi
+      .request(routeData.getUserInfo)
+      .then((data) => {
+        userProfile.setProfileData({
+          userName: data.name,
+          profileDescription: data.about,
+          imageUrl: data.avatar,
+        });
+      })
+      .catch((err) => {
+        console.error(`Failed to get user data: ${err}`);
+      });
+  },
+});
 
-//#region modal handlers
-function checkForModalOverlayClick(evt) {
-    if(evt.target.classList.contains("modal")) onCloseModalClick(evt);
-}
+userProfile.initialize();
+
+const editAvatarForm = new ModalWithForm({
+  modalSelector: modalSelectors.editAvatarSelector,
+  onSubmitCallback: (inputValues) => {
+    function makeRequest() {
+      return dbApi
+        .request(routeData.updateUserAvatar, {
+          avatar: inputValues[userProfileSettings.profileInputAvatarLink],
+        })
+        .then(() => {
+          userProfile.setUserImage(
```

Comment:
* The avatar submit button should be disabled also after the form is reset in `then` https://snipboard.io/7jYl2K.jpg 


`resetValidation` should be called here

Reply:
I moved resetValidation into the handleSubmit function; all forms will now have their validators reset automatically on submit, no need to repeat for each form? Submitting!